integ-test/src/test/java/org/opensearch/sql/sql/TextTypeIT.java

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.sql;
+
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_TEXT;
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_TEXT_FOR_CAST;
+import static org.opensearch.sql.legacy.plugin.RestSqlAction.QUERY_API_ENDPOINT;
+import static org.opensearch.sql.util.MatcherUtils.rows;
+import static org.opensearch.sql.util.MatcherUtils.schema;
+import static org.opensearch.sql.util.MatcherUtils.verifyDataRows;
+import static org.opensearch.sql.util.MatcherUtils.verifySchema;
+import static org.opensearch.sql.util.TestUtils.getResponseBody;
+
+import java.util.Locale;
+import lombok.SneakyThrows;
+import org.json.JSONObject;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Test;
+import org.opensearch.client.Request;
+import org.opensearch.client.RequestOptions;
+import org.opensearch.client.Response;
+import org.opensearch.sql.legacy.SQLIntegTestCase;
+
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+public class TextTypeIT extends SQLIntegTestCase {
+
+  @Override
+  public void init() throws Exception {
+    loadIndex(Index.TEXT);
```

Comment:
Maybe also add tests to check behaviour in functions have not changed. For example, relevance functions unless existing ones are enough.